### PR TITLE
Heterodyne sampling in Fock representation 🔮

### PR DESCRIPTION
**Pull request by sduquemesa**
### Before submitting

Please complete the following checklist when submitting a PR:

- [ ] All new features must include a unit test.
      If you've fixed a bug or added code that should be tested, add a test to the
      test directory!

- [ ] All new functions and code must be clearly commented and documented.
      If you do make documentation changes, make sure that the docs build and
      render correctly by running `make docs`.

- [ ] Ensure that the test suite passes, by running `make test`.

- [ ] Ensure that code and tests are properly formatted, by running `make format` or `black -l 100
      <filename>` on any relevant files. You will need to have the Black code format installed:
      `pip install black`.

- [ ] Add a new entry to the `.github/CHANGELOG.md` file, summarizing the
      change, and including a link back to the PR.

- [ ] The Mr Mustard source code conforms to
      [PEP8 standards](https://www.python.org/dev/peps/pep-0008/) except line length.
      We check all of our code against [Pylint](https://www.pylint.org/).
      To lint modified files, simply `pip install pylint`, and then
      run `pylint mrmustard/path/to/file.py`.

When all the above are checked, delete everything above the dashed
line and fill in the pull request template.

------------------------------------------------------------------------------------------------------------

**Context:**

**Description of the Change:**

**Benefits:**

**Possible Drawbacks:**

**Related GitHub Issues:**

### CODE REVIEW

File: mrmustard/lab/abstract/measurement.py [hunk 1]
```diff
@@ -132,44 +132,48 @@ def _measure_fock(self, other: State) -> Union[State, float]:
...
                 cutoffs.append(
                     max(settings.PNR_INTERNAL_CUTOFF, other.cutoffs[other.indices(mode)])
                 )
-                used += 1
             else:
                 cutoffs.append(other.cutoffs[other.indices(mode)])
-        if self.should_recompute_stochastic_channel() or math.any(
-            [c > settings.PNR_INTERNAL_CUTOFF for c in other.cutoffs]
+        if self.should_recompute_stochastic_channel() or any(
+            c > settings.PNR_INTERNAL_CUTOFF for c in [other.cutoffs[i] for i in self._modes]
```
Comment (ziofil) [inline mrmustard/lab/abstract/measurement.py:152]:
```suggestion
            c > settings.PNR_INTERNAL_CUTOFF for c in [other.cutoffs[other.modes.index(i)] for i in self.modes]
```

File: mrmustard/utils/homodyne.py [hunk 2]
```diff
@@ -241,3 +240,142 @@ def _probs_homodyne_mixed(state_dm, hbar):
...
+    q_tensor = math.new_constant(estimate_quadrature_axis(cutoff), "q_tensor")
+    x = np.sqrt(hbar) * q_tensor
+    p = hbar * q_tensor
+
+    Q, P, probs = husimi_mixed(state_dm.numpy(), x.numpy(), p.numpy(), hbar)
+
+    return Q, P, probs
+
+
+@njit
```
Comment (ziofil) [inline mrmustard/utils/homodyne.py:306]: Maybe we can put husimi in the same place as the new Wigner module? Like a quasiprobability module?

**Review by ziofil — APPROVED** [review]
🚀

File: mrmustard/utils/homodyne.py (hunk 2 above)
Comment (sduquemesa) [reply mrmustard/utils/homodyne.py:306]: done!

**Comment by sduquemesa** [comment]
Closing this stale PR. Note that an implementation of the Husimi function from the Fock representation is implemented here and it is useful for heterodyne sampling; if needed in any situation the code can be reused in a PR that is better integrated with the current architecture of MrMustard.

**Pull request closed without merging**